test/es7/immutable-prototype.js

```diff
@@ -0,0 +1,77 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// ES7 Object Prototype object has an immutable [[Prototype]] internal slot
+// See: 19.1.3 Properties of the Object Prototype Object
+// See: 9.4.7 Immutable Prototype Exotic Objects
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+var tests = [
+    {
+        name: "Okay to set arbitrary object's [[Prototype]] using __proto__",
+        body: function () {
+            var a = {}
+            var b = {'b': 2}
+
+            assert.areEqual(Object.prototype, a.__proto__, "a.__proto__ === Object.prototype")
+            assert.areEqual(Object.prototype, Object.getPrototypeOf(a), "Object.getPrototypeOf(a) === Object.prototype")
+
+            assert.doesNotThrow(function () { eval("a.__proto__ = b") },
+                "It should be okay to set the [[Prototype]] of an arbitrary object using __proto__.")
+
+            assert.areEqual(b, a.__proto__, "a.__proto__ === b")
+            assert.areEqual(b, Object.getPrototypeOf(a), "Object.getPrototypeOf(a) === b")
+        }
+    },
+    {
+        name: "Okay to set arbitrary object's [[Prototype]] using Object.setPrototypeOf",
+        body: function () {
+            var a = {}
+            var b = {'b': 2}
+
+            assert.areEqual(Object.prototype, a.__proto__, "a.__proto__ === Object.prototype")
+            assert.areEqual(Object.prototype, Object.getPrototypeOf(a), "Object.getPrototypeOf(a) === Object.prototype")
+
+            assert.doesNotThrow(function () { eval("Object.setPrototypeOf(a, b)") },
+                "It should be okay to set the [[Prototype]] of an arbitrary object using __proto__.")
+
+            assert.areEqual(b, a.__proto__, "a.__proto__ === b")
+            assert.areEqual(b, Object.getPrototypeOf(a), "Object.getPrototypeOf(a) === b")
+        }
+    },
+    {
+        name: "Not okay to set Object.prototype.[[Prototype]] using __proto__",
+        body: function () {
+            var objectPrototypeObject = Object.getPrototypeOf(Object.prototype)
+            var b = Object.create(null)
+
+            assert.throws(function () { eval("Object.prototype.__proto__ = b") },
```

Comment:
I had only seen assert.throws that used eval. I thought it might have been required for some reason. I don't need it, I'll get rid of it.